jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/globalconfig/GlobalConfigTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.plugins.common.globalconfig;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.cloud.tools.jib.plugins.common.PropertyNames;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.rules.TemporaryFolder;
+
+/** Tests for {@link GlobalConfig}. */
+public class GlobalConfigTest {
+
+  @Rule public final RestoreSystemProperties systemPropertyRestorer = new RestoreSystemProperties();
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private Path configDir;
+
+  @Before
+  public void setUp() {
+    configDir = temporaryFolder.getRoot().toPath();
+  }
+
+  @Test
+  public void testReadConfig_default() throws IOException {
+    GlobalConfig globalConfig = GlobalConfig.readConfig(configDir);
+    assertThat(globalConfig.isDisableUpdateCheck()).isFalse();
+  }
+
+  @Test
+  public void testReadConfig_newConfigCreated() throws IOException {
```

Comment:
`testReadConfig_default` checks the default values when `config.json` doesn't exist. It will check more fields going forward.

`testReadConfig_newConfigCreated` doesn't check default values but checks if `config.json` is created with an expected content.

Reply:
Ah I see.